File: src/api/MintyPeterson.Counter.Api/Models/Requests/EntryNewRequest.cs
// <copyright file="EntryNewRequest.cs" company="Tom Cook">
// Copyright (c) Tom Cook. All rights reserved.
// </copyright>

namespace MintyPeterson.Counter.Api.Models.Requests
{
  /// <summary>
  /// Provides a request model for <see cref="Controllers.EntryController.NewAsync"/>.
  /// </summary>
  public class EntryNewRequest
  {
    /// <summary>
    /// Gets or sets the entry date.
    /// </summary>
    /// <remarks>Only the date part is retained.</remarks>
    public DateTime? EntryDate { get; set; }

    /// <summary>
    /// Gets or sets the entry.
    /// </summary>
    public decimal? Entry { get; set; }

    /// <summary>
    /// Gets or sets the notes.
    /// </summary>
    public string? Notes { get; set; }

    /// <summary>
    /// Gets or sets a value indicating if the entry is an estimate or not.
    /// </summary>
    public bool? IsEstimate { get; set; }
  }
}

File: src/flutter/lib/services/counter/requests/entry_new_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:counter/services/counter/counter_service.dart';

part 'entry_new_request.g.dart';

/// The request model to the [CounterService] entry new endpoint.
@JsonSerializable(createFactory: false)
class EntryNewRequest {

  final DateTime entryDate;
  final int entry;
  final String? notes;
  final bool? isEstimate;
  const EntryNewRequest({
    required this.entryDate,
    required this.entry,
    this.notes,
    this.isEstimate,
  });

  Map<String, dynamic> toJson() => _$EntryNewRequestToJson(this);
}